sso/main.tf

```diff
@@ -0,0 +1,41 @@
+data "aws_caller_identity" "current" {}
+
+module "permission_sets" {
+  source   = "./modules/permission-sets"
+  for_each = { for item in var.associations : item.group => item }
+
+  permission_sets = [
+    {
+      name               = length(each.value.group) < 33 ? "${each.value.group}" : substr("${each.value.group}", 33, length(each.value.group))
```

Comment:
why you cutting only last chunk from string if its length >=33,
cannot we just have the following form of cutting string here and other 2 places so that we will have simple form for all cases?:
`name = substr(each.value.group, 0, 32)`